crates/sui-core/src/unit_tests/move_integration_tests.rs

```diff
@@ -2272,6 +2280,371 @@ async fn test_entry_point_string_option_error() {
+    tests_for_type!(U256, TypeTag::U256, U256::zero());
+    tests_for_type!(SuiAddress, TypeTag::Address, SuiAddress::ZERO);
+    tests_for_type!(
+        ObjectID,
+        TypeTag::Struct(Box::new(sui_types::id::ID::type_())),
+        ObjectID::ZERO
+    );
+    tests_for_type!(&str, utf8_tag(), "❤️🧀");
+    tests_for_type!(&str, ascii_tag(), "love and cheese");
+}
+
+async fn error_test_make_move_vec_for_type<T: Clone + Serialize>(
+    authority: &AuthorityState,
+    gas: &ObjectID,
+    sender: &SuiAddress,
+    sender_key: &AccountKeyPair,
+    t: TypeTag,
+    value: T,
+) {
+    // single without a type argument
+    let mut builder = ProgrammableTransactionBuilder::new();
+    let arg = builder.pure(value.clone()).unwrap();
+    builder.command(Command::MakeMoveVec(None, vec![arg]));
+    let pt = builder.finish();
+    let effects = execute_programmable_transaction(&authority, &gas, &sender, &sender_key, pt)
+        .await
+        .unwrap();
+    assert_eq!(
+        effects.status(),
+        &ExecutionStatus::Failure {
+            error: ExecutionFailureStatus::command_argument_error(
+                CommandArgumentError::TypeMismatch,
+                0
+            ),
+            command: Some(0)
+        }
+    );
+
+    // invalid BCS for any Move value
+    const ALWAYS_INVALID_BYTES: &[u8] = &[255, 255, 255];
+
+    // invalid bcs
+    let mut builder = ProgrammableTransactionBuilder::new();
+    let args = vec![builder.pure_bytes(ALWAYS_INVALID_BYTES.to_vec(), false)];
+    builder.command(Command::MakeMoveVec(Some(t.clone()), args));
+    let pt = builder.finish();
+    let effects = execute_programmable_transaction(&authority, &gas, &sender, &sender_key, pt)
+        .await
+        .unwrap();
+    assert_eq!(
+        effects.status(),
+        &ExecutionStatus::Failure {
+            error: ExecutionFailureStatus::command_argument_error(
+                CommandArgumentError::InvalidBCSBytes,
+                0
+            ),
+            command: Some(0)
+        }
+    );
+
+    // invalid bcs bytes at end
+    let mut builder = ProgrammableTransactionBuilder::new();
+    let args = vec![
+        builder.pure(value.clone()).unwrap(),
+        builder.pure(value.clone()).unwrap(),
+        builder.pure(value).unwrap(),
+        builder.pure_bytes(ALWAYS_INVALID_BYTES.to_vec(), false),
+    ];
+    builder.command(Command::MakeMoveVec(Some(t.clone()), args));
+    let pt = builder.finish();
+    let effects = execute_programmable_transaction(&authority, &gas, &sender, &sender_key, pt)
+        .await
+        .unwrap();
+    assert_eq!(
+        effects.status(),
+        &ExecutionStatus::Failure {
+            error: ExecutionFailureStatus::command_argument_error(
+                CommandArgumentError::InvalidBCSBytes,
+                3,
+            ),
+            command: Some(0)
+        }
+    );
+}
+
+#[tokio::test]
+#[cfg_attr(msim, ignore)]
+async fn test_make_move_vec_errors() {
```

Comment:
This stack overflows for me locally, but if I comment out any single `tests_for_type` it does not... no idea

Reply:
I think I have seen something similar waaay in the past and it went away once tests were split between multiple Rust files (also no idea why and I am not even sure if this is the same issue).